src/Mod/Spreadsheet/Gui/SpreadsheetView.cpp

```diff
@@ -370,6 +371,31 @@ void SheetView::editingFinished()
     }
 }
 
+void SheetView::aliasChanged(const QString& text)
+{
+    // check live the input and highlight if the user input invalid characters
+
+    bool aliasOk = true;
+    QPalette palette = ui->cellAlias->palette();
+
+    if (!text.isEmpty() && !sheet->isValidAlias(Base::Tools::toStdString(text)))
+        aliasOk = false;
+
+    if (!aliasOk) {
+        // change tooltip and make text color red
+        ui->cellAlias->setToolTip(QObject::tr("Alias contains invalid characters!"));

```

Comment:
You should probably remove the exclamation mark. English speakers tend to see it as yelling and offensive.

Reply:
I don't see the problem. The user input an invalid char so it is an error message. In my other software projects it is common to end error messages with an exclamation mark.